middleware/throttle.go

```diff
@@ -34,21 +54,46 @@ type token struct{}
 
 // throttler limits number of currently processed requests at a time.
 type throttle struct {
```

Comment:
for both interfaces and structs ... https://golang.org/src/strings/replace.go?s=305:341#L1,  https://golang.org/src/net/http/server.go?s=1517:2599#L2094, https://golang.org/src/bufio/scan.go?s=1184:1728#L20 etc.